Cql/Cql.Compiler/ExpressionBuilderContext.cs

```diff
@@ -1225,30 +1134,27 @@ protected Expression ParameterRef(ParameterRef op)
     /// <param name="name">The function name</param>
     /// <param name="libraryAlias">If this is an external call, the local alias defined in the using statement</param>
     /// <param name="arguments">The function arguments</param>
+    /// <param name="returnType">The function's return type</param>
     /// <returns></returns>
     protected Expression InvokeDefinedFunctionThroughRuntimeContext(
         string name,
         string? libraryAlias,
-        Expression[] arguments)
+        Expression[] arguments,
+        TypeSpecifier returnType)
     {
         string libraryName = _libraryContext.GetNameAndVersionFromAlias(libraryAlias, throwError: false)
                              ?? throw this.NewExpressionBuildingException($"Local library {libraryAlias} is not defined; are you missing a using statement?");
-
-        var argumentTypes = arguments.SelectToArray(a => a.Type);
-        var selected = _libraryContext.LibraryDefinitions.Resolve(libraryName, name, CheckConversion, argumentTypes);
-        Type definitionType = GetFuncType(selected.Parameters.Select(p => p.Type).Append(selected.ReturnType).ToArray());
-        var parameterTypes = selected.Parameters.Skip(1).Select(p => p.Type).ToArray();
-
-        // all functions still take the bundle and context parameters, plus whatver the operands
-        // to the actual function are.
+        
```

Comment:
The logic here is preventing the demo projects from compiling. I'll have to create an issue to revert it back to the original code, and there is already a ticket for the future to try to get this right